File: rnamake/unittests/secondary_structure_tree_unittest.py
import unittest
import build
from rnamake import secondary_structure_tree, motif_tree

class SecondaryStructureTreeUnittest(unittest.TestCase):

    def test_creation(self):
        sst = secondary_structure_tree.SecondaryStructureTree()

    def test_from_pose(self):
        builder = build.BuildSecondaryStructure()
        ss_p = builder.build_helix(10)
        sst = secondary_structure_tree.tree_from_pose(ss_p)
        if len(ss_p.motifs) != len(sst.tree):
            self.fail("did not get the right number of motifs in tree")

    def test_convert_to_motifs(self):
        builder = build.BuildSecondaryStructure()
        ss_p = builder.build_helix(10)
        sst = secondary_structure_tree.tree_from_pose(ss_p)
        mt = motif_tree.motif_tree_from_ss_tree(sst)
        if len(mt) != 9:
            self.fail("did not get the correct number of nodes in motif tree")

def main():
    unittest.main()

if __name__ == '__main__':
    main()

File: rnamake/lib/RNAMake/unittests/secondary_structure_unittests/secondary_structure_tree_unittest.cpp
//
//  secondary_structure_tree_unittest.cpp
//  RNAMake
//
//  Created by Joseph Yesselman on 12/14/15.
//  Copyright (c) 2015 Joseph Yesselman. All rights reserved.
//

#include "secondary_structure_tree_unittest.h"
#include "build/build_secondary_structure.h"
#include "secondary_structure/secondary_structure_tree.h"



namespace unittests {
namespace sstruct_unittests  {

void
SecondaryStructureTreeUnittest::test_creation() {
    auto sst = sstruct::SecondaryStructureTree();
    
}
    
void
SecondaryStructureTreeUnittest::test_from_pose() {
    auto builder = BuildSecondaryStructure();
    auto p = builder.build_helix();
    auto sst = sstruct::tree_from_pose(p);
}

int
SecondaryStructureTreeUnittest::run() {
    test_creation();
    test_from_pose();
    return 0;
}



}
}